ext/curl/interface.c

```diff
@@ -3811,3 +3954,33 @@ PHP_FUNCTION(curl_pause)
 	RETURN_LONG(curl_easy_pause(ch->cp, bitmask));
 }
 /* }}} */
+
+/* {{{ perform connection upkeep checks */
+PHP_FUNCTION(curl_upkeep)
```

Comment:
if possible, an unit test for this would be nice.

Reply:
I added a **small** test, just to make sure the function is here and get called. Not sure how I can really test this otherwise since it requires a HTTP/2 server.